install/builder.go

```diff
@@ -0,0 +1,119 @@
+package install
+
+import (
+	"context"
+	"github.com/kiegroup/container-builder/client"
+	v08 "github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
+	"github.com/kiegroup/kogito-serverless-operator/utils/kubernetes"
+	"github.com/kiegroup/kogito-serverless-operator/utils/resources"
+	"k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ResourceCustomizer can be used to inject code that changes the objects before they are created.
+type ResourceCustomizer func(object ctrl.Object) ctrl.Object
+
+// IdentityResourceCustomizer is a ResourceCustomizer that does nothing.
+var IdentityResourceCustomizer = func(object ctrl.Object) ctrl.Object {
+	return object
+}
+
+// Resources installs named resources from the project resource directory.
+func Resources(ctx context.Context, c client.Client, namespace string, force bool, customizer ResourceCustomizer, names ...string) error {
+	return ResourcesOrCollect(ctx, c, namespace, nil, force, customizer, names...)
+}
+
+func ResourcesOrCollect(ctx context.Context, c client.Client, namespace string, collection *kubernetes.Collection,
+	force bool, customizer ResourceCustomizer, names ...string) error {
+	for _, name := range names {
+		if err := ResourceOrCollect(ctx, c, namespace, collection, force, customizer, name); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Resource installs a single named resource from the project resource directory.
+func Resource(ctx context.Context, c client.Client, namespace string, force bool, customizer ResourceCustomizer, name string) error {
+	return ResourceOrCollect(ctx, c, namespace, nil, force, customizer, name)
+}
+
+func ResourceOrCollect(ctx context.Context, c client.Client, namespace string, collection *kubernetes.Collection,
+	force bool, customizer ResourceCustomizer, name string) error {
```

Comment:
wouldn't be better to use a struct to define these fields?

Reply:
Are you referring to the function's arguments?

Reply:
Yes.